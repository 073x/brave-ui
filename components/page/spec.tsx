/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import Page from '../page'

describe('page tests', () => {
  const baseComponent = (props?: object) => (
    <Page id='testPage' {...props} />
  )

  it('matches the snapshot', () => {
    const component = baseComponent()
    const tree = create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders the component', () => {
    const wrapper = shallow(baseComponent({id: 'toughPage'}))
    const assertion = wrapper.find('#toughPage').length
    expect(assertion).toBe(1)
  })

  it('can have an id', () => {
    const wrapper = shallow(baseComponent({id: 'boxedThing'}))
    const assertion = wrapper.props().id
    expect(assertion).toBe('boxedThing')
  })

  it('can have a custom title', () => {
    const wrapper = shallow(baseComponent({title: 'here comes the sun'}))
    const assertion = wrapper.props().title
    expect(assertion).toBe('here comes the sun')
  })

  it('can have a custom label', () => {
    const wrapper = shallow(baseComponent({label: 'beta'}))
    const assertion = wrapper.props().label
    expect(assertion).toBe('beta')
  })

  it('can have a custom color', () => {
    const wrapper = shallow(baseComponent({color: 'gray'}))
    const assertion = wrapper.props().color
    expect(assertion).toBe('gray')
  })

  it('can have a custom padding', () => {
    const wrapper = shallow(baseComponent({padding: '10px 20px'}))
    const assertion = wrapper.props().padding
    expect(assertion).toBe('10px 20px')
  })

  it('can have a custom background', () => {
    const wrapper = shallow(baseComponent({background: 'red'}))
    const assertion = wrapper.props().background
    expect(assertion).toBe('red')
  })
})
