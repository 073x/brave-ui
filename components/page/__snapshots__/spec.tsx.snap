// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page tests matches the snapshot 1`] = `
<div
  className="sc-bdVaJa aeEbU"
  color={undefined}
  id="testPage"
  label={undefined}
  title={undefined}
>
  <h1
    className="sc-ifAKCX ipxyF"
    id={undefined}
  />
  <main
    className="sc-bwzfXH glSjfo"
  />
</div>
`;
